DDD-10.md

```diff
@@ -0,0 +1,112 @@
+# DBZ-8968: Add support for pluggable ArrayDeque
+
+## Motivation
+
+Debezium (DBZ) connectors use an in-memory queue (ArrayDeque) within the ChangeEventQueue class to buffer changes between the database reader and the Kafka Connect task. When the Connect task (Eg: IcebergConsumer) is slow to process events, the in-memory queue can become full, causing the database reader to slow down due to backpressure.
+
+For example, The Postgres Connector is single-threaded and memory-bound, which limits throughput. 
+In snapshot phase, DB reads can be parallelized using the `snapshot.fetch.threads` config. However, the benefits are limited without a fast and efficient writer because of limited internal queue capacity.
+
+This behavior introduces limitations, including:
+1. Memory pressure and frequent garbage collection
+2. OOM risks under high load (max.queue.size.in.bytes introduces additional overhead)
+3. Need for vertical scaling (large heap sizes)
+4. Limited throughput for memory bounded connectors (Eg: PostgreSQL)
+
+## Goals
+
+Introduce a pluggable queue abstraction that allows custom implementations of event buffering for
+1. Cost efficiency : Small instances can be used to host the connectors.
```

Comment:
Unfortunately you mostly look at Postgres and you seem to look at it's memory footprints. Did you look at other connectors as well?
Especially when I look at connectors supporting schema history like MySQL connector, I assume instance size cannot be reduced as much as you might expect here. If that's true then I am not convinced if externalizing the queue buffer (alone) brings the expected result. wdyt?

Reply:
I'd also like to add that we have (really long standing) plan to address the schema history memory consumption separately - also maybe adding and SPI for implementations that would combine caching with extrenalization or some schema specific re-use of components.
So I'd see this effort as orthogonal to schema history memory issues and I see it valuable too.